teamcity: failed test: TestReplicaRefreshPendingCommandsTicks

The following tests appear to have failed on master (testrace): TestReplicaRefreshPendingCommandsTicks

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestReplicaRefreshPendingCommandsTicks).

[#1364461](https://teamcity.cockroachdb.com/viewLog.html?buildId=1364461):


```
TestReplicaRefreshPendingCommandsTicks
--- FAIL: testrace/TestReplicaRefreshPendingCommandsTicks (0.100s)

------- Stdout: -------
I190627 21:56:24.319148 27933 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
    replica_test.go:7645: 0: expected no reproposed commands, but found [0xc00123c510]




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -7533,22 +7533,11 @@ func TestReplicaRefreshPendingCommandsTicks(t *testing.T) {
 	defer stopper.Stop(context.TODO())
 	tc.StartWithStoreConfig(t, stopper, cfg)
 
-	// Grab Replica.raftMu in order to block normal raft replica processing. This
-	// test is ticking the replica manually and doesn't want the store to be
-	// doing so concurrently.
 	r := tc.repl
-
 	repDesc, err := tc.repl.GetReplicaDescriptor()
 	if err != nil {
 		t.Fatal(err)
 	}
-
-	// Only followers refresh pending commands during tick events. Change the
-	// replica that the range thinks is the leader so that the replica thinks
-	// it's a follower.
-	r.mu.Lock()
-	r.mu.leaderID = 2
-	r.mu.Unlock()
 
 	electionTicks := tc.store.cfg.RaftElectionTimeoutTicks
 	{

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -7539,6 +7539,15 @@ func TestReplicaRefreshPendingCommandsTicks(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Flush a write all the way through the Raft proposal pipeline. This
+	// ensures that leadership settles down before we start manually submitting
+	// proposals and that we don't see any unexpected proposal refreshes due to
+	// reasons like reasonNewLeaderOrConfigChange.
+	args := incrementArgs([]byte("a"), 1)
+	if _, pErr := tc.SendWrapped(&args); pErr != nil {
+		t.Fatal(pErr)
+	}
+
 	electionTicks := tc.store.cfg.RaftElectionTimeoutTicks
 	{
 		// The verifications of the reproposal counts below rely on r.mu.ticks

